tools/zetasql-helper/bigquery-antipattern-recognition/src/main/java/com/google/zetasql/toolkit/antipattern/parser/visitors/singlerowinsert/IdentifySingleRowInsertsVisitor.java

```diff
@@ -0,0 +1,42 @@
+package com.google.zetasql.toolkit.antipattern.parser.visitors.singlerowinsert;
+
+import com.google.common.collect.ImmutableList;
+import com.google.zetasql.parser.ASTNodes;
+import com.google.zetasql.parser.ParseTreeVisitor;
+import java.util.ArrayList;
+
+public class IdentifySingleRowInsertsVisitor extends ParseTreeVisitor {
+
+    private final static String SINGLE_ROW_INSERTS = "SINGLE ROW INSERTS";
+    private final static String OTHER_INSERT_PATTERN = "OTHER INSERT PATTERN";
+    private ArrayList<String> result = new ArrayList<String>();
+
+    public ArrayList<String> getResult() {
+        return result;
+    }
+
+
+    @Override
+    public void visit(ASTNodes.ASTInsertValuesRowList node) {
+        ImmutableList<ASTNodes.ASTInsertValuesRow> nodes = node.getRows();
+        if(nodes.size() == 1 ){
+            result.add(String.format(SINGLE_ROW_INSERTS));
+        }
+        else{
+            result.add(String.format(OTHER_INSERT_PATTERN));
```

Comment:
For now we will only be reporting SINGLE_INSERT_PATTERNS. Please remove the OTHER_INSERT_PATTERN.

Reply:
Thanks for reviewing the logic, I have removed the OTHER_INSERT_PATTERN part
Please check and let me know